Release How-To & Conda

ax3l opened the pull request:
Adds a release how-to and conda docker container.

Since I copied and modified it from openPMD-viewer, @RemiLehe  is set as the author of the first commit via:
```bash
GIT_AUTHOR_NAME="Remi Lehe" GIT_AUTHOR_EMAIL="remi.lehe@normalesup.org" git commit
```

RemiLehe committed:
Release How-To & Conda

Adds a release how-to and conda docker container.

Since I copied and modified it from openPMD-viewer, Remi is set
as the author of this commit.

ax3l commented on conda_recipe/Dockerfile:
```diff
@@ -0,0 +1,22 @@
+FROM continuumio/miniconda
+MAINTAINER Axel Huebl <a.huebl@hzdr.de>
+
+# RUN apt-get update \
```
to do: test and remove if not needed

ax3l replied:
indeed not needed here :)

ax3l commented on conda_recipe/meta.yaml:
```diff
@@ -0,0 +1,36 @@
+{% set version = "1.0.0" %}
```
I would add a *fourth* version number here so that we can make releases of the validator that are always the same major.minor.patch as the openPMD standard but still have a level for improvements and bug fixes within the same standard version.

Should be possible, right?

ax3l replied:
hm, pbr does have troubles with that although PEP-440 supports it...

ax3l replied:
but reported under: https://bugs.launchpad.net/pbr/+bug/1734361

otherwise we can still use vanilla...

ax3l committed:
Update release notes further

ax3l commented:
I might switch back to the vanilla installer since `pbr` has a bug on valid, 4-digit versions:
- https://github.com/openPMD/openPMD-validator/pull/26#discussion_r153003390
- https://github.com/openPMD/openPMD-validator/pull/22#issuecomment-346870270

t184256 commented:
Hm, that's unfortunate. Sorry for wasting your time on fighting `pbr` when you didn't really even need it. #23?..

ax3l commented:
Don't worry, no harm done :)

I had a local draft already but if Rémi wants to provide a setup.py I would also welcome PR :)

If you are ok, I can take care of the releases for this repo (pypi, conda, spack).

RemiLehe commented:
@ax3l I updated the `setup.py` from @t184256's PR #23  (Thanks for providing it, @t184256, and no worries: I am the one to blame for merging #22 too quickly), and created a corresponding PR (#27). 

One #27 is merged, you could rebase the current PR.

ax3l committed:
Conda: Finalize Build

- linux-64 (132K)
- linux-ppc64le (32M)
- osx-64 (2.4M)
- win-64 (617M)

ax3l commented:
@RemiLehe I finished the PR, merg-able :)

ax3l committed:
README: More Badges

Fix markdown markup on pypi and add badges for pypi, conda,
pyversions and license

ax3l committed:
Twine: Sign Release

ax3l committed:
Badge: Anaconda Cloud

RemiLehe commented on RELEASING.md:
```diff
@@ -0,0 +1,91 @@
+
+- you should install the packages
+[`pypandoc`](https://pypi.python.org/pypi/pypandoc/),
+[`twine`](https://pypi.python.org/pypi/twine), [`pbr`](https://pypi.python.org/pypi/pbr)
```
pbr is not needed anymore

RemiLehe commented:
Thanks for this PR. I have only a minor comment (see above)

ax3l committed:
Final Updates

- we do not need pbr anymore
- fix final directory rights

ax3l commented:
@RemiLehe thank you, fixed it! :)
